frontend/nexusInteraction.cpp

```diff
@@ -420,4 +424,185 @@ void MainWindow::showStatus(qreal x, qreal y, QString title)
+
+        gridLayout->addWidget(chartView, 1, 0, -1, -1);
+        QString tabName = "Spectra";
+        ui_->tabWidget->addTab(window, tabName);
+        ui_->tabWidget->setCurrentIndex(ui_->tabWidget->count() - 1);
+        runs.chop(2);
+        QString toolTip = field + "\n" + runs;
+        ui_->tabWidget->setTabToolTip(ui_->tabWidget->count() - 1, toolTip);
+        chartView->setFocus();
+    }
+    else
+    {
+        // an error occurred
+        msg = "Error2: " + worker->error_str;
+        QMessageBox::information(this, "", msg);
+    }
+}
+
+void MainWindow::getSpectraCount()
+{
+    // Gathers all selected runs
+    auto selectedRuns = ui_->runDataTable->selectionModel()->selectedRows();
+    // Finds run number location in table
+    int runNoColumn;
+    for (auto i = 0; i < ui_->runDataTable->horizontalHeader()->count(); ++i)
+    {
+        if (model_->headerData(i, Qt::Horizontal, Qt::UserRole).toString() == "run_number")
+        {
+            runNoColumn = i;
+            break;
+        }
+    }
+    // Gets all selected run numbers and fills graphing toggles
+    QString runNos = "";
+    QString runNo;
+    // Concats runs
+    for (auto run : selectedRuns)
+    {
+        runNo = model_->index(run.row(), runNoColumn).data().toString();
+        if (runNo.contains("-") || runNo.contains(","))
+        {
+            QString groupedRuns;
+            auto runArr = runNo.split(",");
+            foreach (const auto &string, runArr)
+            {
+                if (string.contains("-"))
+                {
+                    for (auto i = string.split("-")[0].toInt(); i <= string.split("-")[1].toInt(); i++)
+                        groupedRuns += QString::number(i) + ";";
+                }
+                else
+                    groupedRuns += string + ";";
+            }
+            groupedRuns.chop(1);
+            runNos.append(groupedRuns + ";");
+        }
+        else
+            runNos.append(runNo + ";");
+    }
+    // Removes final ";"
+    runNos.chop(1);
+    // Error handling
+    if (runNos.size() == 0)
+        return;
+
+    QString cycle = cyclesMap_[ui_->cycleButton->text()];
+    cycle.replace(0, 7, "cycle").replace(".xml", "");
+
+    QString url_str = "http://127.0.0.1:5000/getSpectra/";
+    url_str += instName_ + "/" + cycle + "/" + runNos;
+    HttpRequestInput input(url_str);
+    auto *worker = new HttpRequestWorker(this);
+    connect(worker, SIGNAL(on_execution_finished(HttpRequestWorker *)), this, SLOT(plotSpectra(HttpRequestWorker *)));
+    setLoadScreen(true);
+    worker->execute(input);
+}
+void MainWindow::plotSpectra(HttpRequestWorker *count)
+{
+    setLoadScreen(false);
+    auto spectraCount = count->response.toUtf8();
+    spectraCount.chop(1);
+    auto spectrumNumber =
+        QInputDialog::getInt(this, tr("Plot Detector Spectrum"),
+                             tr("Enter detector spectrum to plot (0-" + spectraCount + "):"), 0, 0, count->response.toInt(), 1);
```

Comment:
Arrays in the NeXuS file are zero-indexed I believe, so the max value should be `count->response.toInt() - 1`